utility/unit_of_work/transaction_test.go

```diff
@@ -7,69 +7,68 @@ import (
 
 	"github.com/pokt-network/pocket/runtime/test_artifacts"
 	"github.com/pokt-network/pocket/shared/codec"
-	"github.com/pokt-network/pocket/shared/core/types"
 	coreTypes "github.com/pokt-network/pocket/shared/core/types"
 	"github.com/pokt-network/pocket/shared/crypto"
 	"github.com/pokt-network/pocket/shared/utils"
-	typesUtil "github.com/pokt-network/pocket/utility/types"
 	"github.com/stretchr/testify/require"
 )
 
 var (
 	defaultSendAmount = big.NewInt(10000)
 )
 
-func TestUtilityContext_AnteHandleMessage(t *testing.T) {
-	ctx := newTestingUtilityContext(t, 0)
+func TestUtilityUnitOfWork_AnteHandleMessage(t *testing.T) {
+	uow := newTestingUtilityUnitOfWork(t, 0)
 
-	tx, startingBalance, _, signer := newTestingTransaction(t, ctx)
-	msg, err := ctx.anteHandleMessage(tx)
+	tx, startingBalance, _, signer := newTestingTransaction(t, uow)
+	msg, err := uow.anteHandleMessage(tx)
 	require.NoError(t, err)
 	require.Equal(t, signer.Address().Bytes(), msg.GetSigner())
-	feeBig, err := ctx.getMessageSendFee()
+	feeBig, err := uow.getMessageSendFee()
 	require.NoError(t, err)
 
 	expectedAfterBalance := big.NewInt(0).Sub(startingBalance, feeBig)
-	amount, err := ctx.getAccountAmount(signer.Address())
+	amount, err := uow.getAccountAmount(signer.Address())
 	require.NoError(t, err)
 	require.Equal(t, expectedAfterBalance, amount, "unexpected after balance")
 }
 
-func TestUtilityContext_ApplyTransaction(t *testing.T) {
-	ctx := newTestingUtilityContext(t, 0)
+func TestUtilityUnitOfWork_ApplyTransaction(t *testing.T) {
+	uow := newTestingUtilityUnitOfWork(t, 0)
 
-	tx, startingBalance, amount, signer := newTestingTransaction(t, ctx)
-	txResult, err := ctx.hydrateTxResult(tx, 0)
+	tx, startingBalance, amount, signer := newTestingTransaction(t, uow)
+	txResult, err := uow.hydrateTxResult(tx, 0)
 	require.NoError(t, err)
 	require.Equal(t, int32(0), txResult.GetResultCode())
 	require.Equal(t, "", txResult.GetError())
-	feeBig, err := ctx.getMessageSendFee()
+	feeBig, err := uow.getMessageSendFee()
 	require.NoError(t, err)
 
 	expectedAmountSubtracted := amount.Add(amount, feeBig)
 	expectedAfterBalance := big.NewInt(0).Sub(startingBalance, expectedAmountSubtracted)
-	amount, err = ctx.getAccountAmount(signer.Address())
+	amount, err = uow.getAccountAmount(signer.Address())
 	require.NoError(t, err)
 	require.Equal(t, expectedAfterBalance, amount, "unexpected after balance")
 }
 
-func TestUtilityContext_HandleTransaction(t *testing.T) {
-	ctx := newTestingUtilityContext(t, 0)
-	tx, _, _, _ := newTestingTransaction(t, ctx)
+// TODO: - @deblasis - refactor this to test HandleTransaction specifically in the utility package
```

Comment:
Is this in a followup PR?

Reply:
yes, all these TODOes are gonna be removed within this set of PRs